### PR TITLE
feat: Add Optional field in ServiceConfig struct

### PR DESCRIPTION
**Pull request by cloudxxx8**
For more configuration providers in the future, add an extendable field in the ServiceConfig struct to make the extension easier.
Fix https://github.com/edgexfoundry/go-mod-configuration/issues/74

Signed-off-by: Cloud Tsai <cloudxxx8@gmail.com>

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-configuration/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) N/A, a new filed only
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?) 
- [ ] I have opened a PR for the related docs change (if not, why?) No, only add the godoc in the code

## Testing Instructions
N/A, no behavior change

### CODE REVIEW

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by cloudxxx8** [comment]
@lenny-intel please help review again.  I use go 1.18 `any` to replace `interface{}` here.

File: pkg/types/config.go [hunk 1]
```diff
@@ -45,6 +45,9 @@ type ServiceConfig struct {
 	// GetAccessToken is a callback function that retrieves a new Access Token.
 	// This callback is used when a '403 Forbidden' status is received from any call to the configuration provider service.
 	GetAccessToken GetAccessTokenCallback
+	// Optional contains all other properties of the configuration provider might use.
+	// For example, it might need the message bus connection information to publish the config changes.
+	Optional map[string]any
```
Comment (lenny-intel) [inline pkg/types/config.go:50]: Love `any` !!

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**